### PR TITLE
Fix `core:unicode/utf8.decode_grapheme_iterate` returning wrong text

### PR DESCRIPTION
**Pull request by blob1807**
The current behavior uses the monospace width of a grapheme, which isn't the same as the data width.

This PR also fixes a typo & adds documentation to `Grapheme.width`, to help people not assume it's the data width. Though it should be renamed to something more explicate like `monospace_width` or `cell_count`, but I couldn't decided which is better.

Example of current behavior:
```odin
package __test_bug
 
import "core:fmt"
import "core:unicode/utf8"

main :: proc() { 
    base := "子猫"
    iter := utf8.decode_grapheme_iterator_make(base)
    for str, g in utf8.decode_grapheme_iterate(&iter) {
        fmt.println(str, transmute([]u8)str, g)
    }
}
```
This should give:
```
子 [229, 173, 144] Grapheme{byte_index = 0, rune_index = 0, width = 2}
猫 [231, 140, 171] Grapheme{byte_index = 3, rune_index = 1, width = 2}
```
but instead gives (output of `str` may be nothing depending on your term):
```
σ¡ [229, 173] Grapheme{byte_index = 0, rune_index = 0, width = 2}
τî [231, 140] Grapheme{byte_index = 3, rune_index = 1, width = 2}
```

### CODE REVIEW

File: core/unicode/utf8/grapheme.odin [hunk 1]
```diff
@@ -23,7 +23,7 @@ normalized_east_asian_width       :: unicode.normalized_east_asian_width
 Grapheme :: struct {
 	byte_index: int,
 	rune_index: int,
-	width:      int,
+	width:      int, // The width of the Grapheme in number of monospace cells
```
Comment (Kelimion) [inline core/unicode/utf8/grapheme.odin:26]: I'd go with `cell_count: int`.

Comment (blob1807) [reply core/unicode/utf8/grapheme.odin:26]: Should I also do the same for the other uses of `width` or are those clear enough?

**Comment by Kelimion** [comment]
Good find. Can you also add your example test case to `tests/core/unicode/test_core_unicode.odin`?

It has a number of string -> expected grapheme tests already, but it appears their "width" and "width" must've been the same for this not to be caught.

File: core/unicode/utf8/grapheme.odin (hunk 1 above)
Comment (Kelimion) [reply core/unicode/utf8/grapheme.odin:26]: On second thought, let's stick with `width` with the added comment. It's clear with that added context, and with that clarification, there is no need to break existing code.

So let's just add the comment to `Grapheme.width` and fix `Grapheme_Iterator`'s typo for the same.

**Comment by blob1807** [comment]
Sure I can do that.

**Comment by Kelimion** [comment]
Thanks!

**Comment by blob1807** [comment]
I'm going to close this PR, as I assumed `text` was meant to be the raw text of the grapheme. 
However during making the test, I realized the algorithm doesn't fully decode the current grapheme. Instead it decodes the last one & returns the start of the next. 
So it could never be the raw text of the grapheme, but the just start of the current one. Which is weird to me.

I just lucked out in the example I gave.

**Comment by Kelimion** [comment]
Understood. And now we know. :-)
